### PR TITLE
Initial addition of opensearch highlights to graphql

### PR DESCRIPTION
**Pull request by JPrevost**
#### Why these changes are being introduced:

Since we show minimal metadata in the brief results, it might be
useful for end users to see where in the metadata their search terms
matched.

#### Relevant ticket(s):

https://mitlibraries.atlassian.net/browse/RDI-156

#### How this addresses that need:

This enables the OpenSearch feature to highlight query matches and
updates the GraphQL configuration to return that in the results.
Because we target `_source` as our GraphQL records, and the
highlights appear in `hits`, we inject the field into each record
in the QueryType class.

This approach also allows to return `score` in GraphQl results. Because
`score` will be useful to us in the near future, this commit configures
that as well.

#### Side effects of this change:

* We will likely want to reevaluate how we do this after removing
the deprecated V1 fields. I've added a comment to remind of this to
QueryType#inject_hits_fields_into_source, but we should also open a
JIRA ticket once we confirm that this approach is acceptable in the
interim.
* Highlights are returned as an array of strings, so we'll need to
transform this to unserialized JSON in TIMDEX-UI.
* Some unrelated test assertions had to be updated, presumably
because more data sources had been loaded since the last time
cassettes were regenerated. If this becomes burdensome, we might
want to consider using staging OpenSearch and filtering secrets.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

**Review by matt-bernhardt — APPROVED** [review]
This looks fine to me, and I agree with the strategy of making a ticket to streamline this implementation after the v1 implementation has been removed.

I asked one question about the `inject_hits_fields_into_sources` method, but only to understand a detail of the approach taken - I don't think it needs to change.

So long as Helen signs off on this from the data model perspective, I think this is okay from the code and testing side.

File: app/graphql/types/query_type.rb [hunk 1]
```diff
@@ -89,10 +89,24 @@ def search(searchterm:, citation:, contributors:, funding_information:, identifi
...
         response
       end
+
+      # Long-term, we will probably want to define these fields discretely in RecordType. However, this might end up
+      # adding confusion while we are maintaining deprecated fields in that class. We should refactor this either soon
+      # after removing GraphQL V1 or as part of that work.
+      def inject_hits_fields_into_source(hits)
+        modded_source = []
+        hits.each do |hit|
+          source = hit['_source']
```
[the remark this answers is not in the document]
Comment (JPrevost) [reply app/graphql/types/query_type.rb:103]: modded_source is an array of modified `source`s. It's possible `modded_sources` would be a better name to make that more clear.

File: app/graphql/types/record_type.rb [hunk 2]
```diff
@@ -126,6 +126,19 @@ class LocationType < Types::BaseObject
       field :kind, String
       field :value, String
     end
+
+    class HighlightType < Types::BaseObject
+      field :kind, String
```
Comment (JPrevost) [inline app/graphql/types/record_type.rb:131]: I wonder if `:field` or `:matched_field` would be more useful here than `:kind`?

File: app/graphql/types/record_type.rb [hunk 3]
```diff
@@ -126,6 +126,19 @@ class LocationType < Types::BaseObject
...
+      field :value, [String]
```
Comment (JPrevost) [inline app/graphql/types/record_type.rb:132]: Similarly, I wonder if `:matched_phrases` or something more descriptive than `:value` would be more useful...

**Review by hakbailey — APPROVED** [review]
Looks great!

**Pull request merged**